File: Session_03_Selection/selection.py
# 1. If-Elif-Else structure
score = 78
print("--- Grade check (if-elif-else) ---")
print(f"Score: {score}")

if score >= 85:
    print("Grade: A")
elif score >= 75: # Note Python uses 'elif' instead of 'else if'
    print("Grade: B")
elif score >= 65:
    print("Grade: C")
else:
    print("Grade: F (Fail)")

# 2. Python has match-case (since Python 3.10) equivalent to switch-case
menu_choice = 2
print("\n--- System Menu (match-case) ---")
print(f"Choice: {menu_choice}")

match menu_choice:
    case 1:
        print("Running program: Add Student Records")
    case 2:
        print("Running program: View Student Grades") // No break keyword needed in Python!
    case 3:
        print("Running program: Exit System")
    case _: # Underscore represents default case
        print("Error: Invalid choice!")

File: Session_03_Selection/selection.c
#include <stdio.h>

int main() {
    // 1. If-Else structure
    int score = 78;
    printf("--- Grade check (if-else) ---\n");
    printf("Score: %d\n", score);

    if (score >= 85) {
        printf("Grade: A\n");
    } else if (score >= 75) {
        printf("Grade: B\n");
    } else if (score >= 65) {
        printf("Grade: C\n");
    } else {
        printf("Grade: F (Fail)\n");
    }

    // 2. Switch-Case structure (Perfect for menus!)
    int menu_choice = 2;
    printf("\n--- System Menu (switch-case) ---\n");
    printf("Choice: %d\n", menu_choice);

    switch (menu_choice) {
        case 1:
            printf("Running program: Add Student Records\n");
            break;
        case 2:
            printf("Running program: View Student Grades\n");
            break; // Exits switch block safely
        case 3:
            printf("Running program: Exit System\n");
            break;
        default:
            printf("Error: Invalid choice!\n");
            break;
    }

    return 0;
}
